aip/client-libraries/4235.md

```diff
@@ -0,0 +1,59 @@
+---
+id: 4235
+state: draft
+created: 2024-02-15
+---
+
+# Automatically populate fields in the request message
+
+For APIs that leverage request idempotency as described via [AIP-155], it could
+be useful to have the client libraries automatically populate the `request_id`
+if it is
+not already set by the customer.
+
+**Note:** This feature is primarily written for `request_id` fields within the
+request message. However, this feature should work for a field of any name, as
+long as the below conditions hold true.
+
+## Guidance
+
+APIs **may** configure fields in the request message for automatic population.
+
+For a field to be automatically populated, the below configurations **must** be
+true:
+
+- The field **must** be of type `string`
+- The field **must** be at the top-level of the request message (e.g. nested
+  fields are not supported)
```

Comment:
I feel it may not be necessary to mention the cases not covered by the AIP, but if that is a common practice for AIPs, please disregard this comment.

Reply:
That's a fair point - I added the comment more to help clarify what "top-level of request message" means, rather than to mention it as a case not covered by the AIP. 
But I checked a couple other AIPs that use the term `top-level`, and it seems commonly used enough that I don't think I need this extra clarification, so I've removed it.